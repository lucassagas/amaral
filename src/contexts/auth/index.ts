export { AuthContext, AuthContextProvider, handleSignOut } from './auth'
