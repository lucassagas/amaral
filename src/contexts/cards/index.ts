export { CardContext, CardContextProvider } from './cards';

