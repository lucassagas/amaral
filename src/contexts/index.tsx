import { theme } from '@/styles/theme'
import { ChakraProvider } from '@chakra-ui/react'
import { ReactNode } from 'react'
import { QueryClientProvider } from 'react-query'
import { ReactQueryDevtools } from 'react-query/devtools'
import { queryClient } from 'src/lib/queryClient'
import { AuthContextProvider } from './auth'

type AppProviderProps = {
  children: ReactNode
}

export function AppProvider({ children }: AppProviderProps) {
  return (
    <ChakraProvider theme={theme}>
      <AuthContextProvider>
        <QueryClientProvider client={queryClient}>
          {children}
          <ReactQueryDevtools />
        </QueryClientProvider>
      </AuthContextProvider>
    </ChakraProvider>
  )
}
