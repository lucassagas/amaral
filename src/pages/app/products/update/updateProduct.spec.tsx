import { AppProvider } from '@/contexts/index'
import { render, screen } from '@testing-library/react'
import UpdateProduct from './[id]'

const mockedProduct = {
  id: 'uuid',
  name: 'Camarao',
  quantity: 200,
  createdAt: new Date(),
  updatedAt: new Date(),
}

jest.mock('next/router', () => ({
  useRouter() {
    return {
      route: '',
      pathname: 'example.com',
      query: '',
      asPath: 'example.com',
    }
  },
}))

const renderComponent = () => {
  return render(
    <AppProvider>
      <UpdateProduct product={mockedProduct} />
    </AppProvider>
  )
}

describe('<UpdateProduct />', () => {
  it('should render correctly', () => {
    const { container } = renderComponent()

    screen.logTestingPlaygroundURL()

    expect(container).toMatchSnapshot()
  })
})
