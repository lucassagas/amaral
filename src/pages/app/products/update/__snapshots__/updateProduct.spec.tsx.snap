// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UpdateProduct /> should render correctly 1`] = `
<div>
  <div
    class="css-qqp9b6"
  >
    <h2
      class="chakra-heading css-sxvlx"
    >
      Editar Produto
    </h2>
    <hr
      aria-orientation="horizontal"
      class="chakra-divider css-e44dav"
    />
    <form
      class="css-0"
    >
      <div
        class="chakra-stack css-uc6l60"
      >
        <div
          class="css-14zyk4t"
        >
          <div
            class="chakra-form-control css-1kxonj9"
            role="group"
          >
            <label
              class="chakra-form__label css-pk5tju"
              for="email"
              id="field-1-label"
            >
              Nome
            </label>
            <input
              class="chakra-input css-uxk25m"
              id="name"
              name="name"
              value="Camarao"
            />
          </div>
          <div
            class="chakra-form-control css-1kxonj9"
            role="group"
          >
            <label
              class="chakra-form__label css-pk5tju"
              for="email"
              id="field-2-label"
            >
              Quantidade
            </label>
            <input
              class="chakra-input css-uxk25m"
              id="quantity"
              name="quantity"
              value="200"
            />
          </div>
        </div>
      </div>
      <div
        class="css-bxm9v3"
      >
        <div
          class="chakra-stack css-a9v878"
        >
          <a
            class="chakra-button css-r8khcr"
            href="/app/products"
          >
            Cancelar
          </a>
          <button
            class="chakra-button css-mwvyhu"
            type="submit"
          >
            Salvar
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;
